src/engine/controls/enginecontrol.cpp

```diff
@@ -115,3 +117,11 @@ EngineBuffer* EngineControl::pickSyncTarget() {
     }
     return nullptr;
 }
+
+EngineControl::FrameOfTrack EngineControl::getFrameOfTrack() const {
+    FrameOfTrack frameOfTrack{
+            samplePosToFramePos(getSampleOfTrack().current),
```

Comment:
You must not call getSampleOfTrack() twice in a single function,
else the values might be unrelated due to a write in between. 

In this case you should make a local copy first.